src/components/Register/Register.tsx

```diff
@@ -1,62 +1,103 @@
 import React from "react";
-import Button from "@mui/material/Button";
+import Dialog from "src/components/Dialog";
+import { supabase } from "src/utils/supabaseClient";
 import useFormConfig from "@src/hooks/useFormConfig";
 import { FieldProps } from "@src/hooks/useFormConfig/useFormConfig";
 import TextInput from "../TextInput";
-import { Styledform } from "./styles";
+import { StyledButton, Styledform } from "./styles";
 
 const fields: FieldProps[] = [
   {
-    id: "firstName",
+    id: "first_name",
     initialValue: "",
-    placeholder: "",
+    placeholder: "John",
     label: "First name",
     validate: "text",
     required: true
   },
   {
-    id: "lastName",
+    id: "last_name",
     initialValue: "",
-    placeholder: "",
+    placeholder: "Doe",
     label: "Last name",
     validate: "text",
     required: true
   },
   {
     id: "email",
     initialValue: "",
-    placeholder: "",
+    placeholder: "youremail@domain.com",
     label: "Email",
     validate: "email",
     required: true
   },
   {
-    id: "phone_number",
+    id: "password",
     initialValue: "",
-    placeholder: "",
-    label: "Phone number",
-    validate: "phone_number",
+    placeholder: "********",
+    label: "Password",
+    validate: "password",
     required: true
   }
 ];
 
 function Register() {
-  const formik = useFormConfig(fields);
+  const handleSubmit = async (values: any) => {
+    const { email, password, firstName, lastName } = values;
+    const isNewUser = await supabase.isNewUSer(email);
+    if (isNewUser) {
+      const response = await supabase.auth.signUp(
+        {
+          email,
+          password
+        },
+        {
+          data: {
+            first_name: firstName,
+            last_name: lastName
+          }
```

Comment:
Noté que el link que llega al correo apunta para localhost cuando deberia de apuntar a el dominio. Para esto se puede usar el parametro `redirectTo`
https://supabase.com/docs/reference/javascript/auth-signup#parameters

```
redirectTo: `${window.location.origin}`
```